dash-pipeline/SAI/templates/utils.cpp.j2

```diff
@@ -217,20 +217,86 @@ sai_status_t sai_create_switch_dummy(
     return SAI_STATUS_SUCCESS;
 }
 
-sai_status_t sai_get_switch_attribute_dummy(
+#define DASH_BMV2_NUM_PORTS 2 
+#define DASH_BMV2_CPU_QOS_NUMBER_OF_QUEUES 0
+#define DASH_BMV2_DEFAULT_CPU_PORT_ID (sai_object_id_t)((SAI_OBJECT_TYPE_PORT<<48)+1)
+#define DASH_BMV2_DEFAULT_VLAN_ID (sai_object_id_t)((SAI_OBJECT_TYPE_VLAN<<48)+1)
+#define DASH_BMV2_DEFAULT_VRF_ID (sai_object_id_t)((SAI_OBJECT_TYPE_VIRTUAL_ROUTER<<48)+1)
+#define DASH_BMV2_DEFAULT_1Q_BRIDGE_ID (sai_object_id_t)((SAI_OBJECT_TYPE_BRIDGE<<48)+1)
+#define DASH_BMV2_DEFAULT_1Q_BRIDGE_ID (sai_object_id_t)((SAI_OBJECT_TYPE_BRIDGE<<48)+1)
+
+sai_object_id_t port_list[DASH_BMV2_NUM_PORTS] = {
+    (sai_object_id_t)((SAI_OBJECT_TYPE_PORT<<48) + 1),
+    (sai_object_id_t)((SAI_OBJECT_TYPE_PORT<<48) + 2)
+};
+
+sai_status_t sai_get_switch_attribute(
         _In_ sai_object_id_t switch_id,
         _In_ uint32_t attr_count,
         _Inout_ sai_attribute_t *attr_list) {
 
-    fprintf(stderr, "sai_get_switch_attribute_dummy()\n");
+    fprintf(stderr, "sai_get_switch_attribute()\n");
+    int i;
+    sai_attribute_t *attr = attr_list;
+    sai_object_list_t port_obj_list;
+    sai_object_id_t *objlist;
+    for (i = 0; i < attr_count ; i++, attr++) {
+        switch(attr->id) {
+        
+        case SAI_SWITCH_ATTR_NUMBER_OF_ACTIVE_PORTS:
+            attr->value.u32 = DASH_BMV2_NUM_PORTS;
+            fprintf(stderr, "  sai_get_switch_attribute() [%d] attr %d SAI_SWITCH_ATTR_NUMBER_OF_ACTIVE_PORTS = %d\n",
+                    i, attr->id, attr->value.u32);
+            return SAI_STATUS_SUCCESS;
+        
+        case SAI_SWITCH_ATTR_PORT_LIST:
+            // make a tmp port list, saiserver will free the memory
+            objlist = (sai_object_id_t *)malloc(sizeof(port_list));
```

Comment:
I think we can use cpp style allocation instead of C malloc

Reply:
If you have a specific suggestion I'll consider it, otherwise I'm happy with this since it's plain c-code in any case.